socorro/external/postgresql/setupdb_app.py

```diff
@@ -268,75 +290,88 @@ def create_table(element, compiler, **kw):
-                                   self.database_name)
-            except exc.ProgrammingError, e:
-                if re.search(
-                    'database "%s" already exists' % self.database_name,
-                    e.orig.pgerror.strip()):
-                    # already done, no need to rerun
-                    print "The DB %s already exists" % self.database_name
-                    return 0
-                raise
-
-            db.create_roles(self.config)
-            connection.close()
-
-        # Reconnect to set up schema, types and procs
-        sa_url = url_template + '/%s' % self.database_name
         alembic_cfg = Config(self.config.alembic_config)
-        alembic_cfg.set_main_option("sqlalchemy.url", sa_url)
-        with PostgreSQLAlchemyManager(sa_url, self.config.logger) as db:
-            connection = db.engine.connect()
-            db.setup_admin()
-            if self.no_schema:
-                db.commit()
-                return 0
-            # Order matters with what follows
-            db.create_types()
-
-            db.create_procs()
-            db.set_sequence_owner('breakpad_rw')
+        alembic_cfg.set_main_option('sqlalchemy.url', connection_url)
+
+        with PostgreSQLAlchemyManager(connection_url, self.config.logger,
+                                      False, self.on_heroku) as db:
+            db.setup_extensions()
+            db.grant_public_schema_ownership(database_username)
+            db.commit()
+
+        if self.no_schema:
+            return 0
+
+        # Reconnect as a regular user to set up schema, types and procs
+        if has_database_url:
+            connection_url = self.create_connection_url(
+                database_name,
+                database_username,
+                database_password
+            )
+
+        with PostgreSQLAlchemyManager(connection_url, self.config.logger,
+                                      False, self.on_heroku) as db:
+            # Order matters below
+            db.turn_function_body_checks_off()
+            db.load_raw_sql('types')
+            db.load_raw_sql('procs')
+            # We need to commit to make a type visible for table creation
             db.commit()
 
             db.create_tables()
-            db.set_table_owner('breakpad_rw')
-            db.create_views()
+            db.load_raw_sql('views')
             db.commit()
 
-            db.set_grants(self.config)  # config has user lists
-            if not self.config['no_staticdata']:
+            if not self.config.get('no_staticdata'):
                 self.import_staticdata(db)
-            if self.config['fakedata']:
-                self.generate_fakedata(db, self.config['fakedata_days'])
+            if self.config.get('fakedata'):
+                self.generate_fakedata(db, self.config.get('fakedata_days'))
             db.commit()
             command.stamp(alembic_cfg, "heads")
```

Comment:
at this point in the code, the ```alembic_cfg``` object has reference to a different ```connection_url``` than the ```PostgreSQLAlchemyManager``` context.  Is this correct?